### PR TITLE
build: update go version to 1.18

### PR DESCRIPTION
**Pull request by asraa**
Signed-off-by: Asra Ali <asraa@google.com>

Please fill in the fields below to submit a pull request.  The more information that is provided, the better.

Fixes https://github.com/theupdateframework/go-tuf/issues/307
Release Notes: 

**Types of changes**:

- [  ] Bug fix (non-breaking change which fixes an issue)
- [  ] New feature (non-breaking change which adds functionality)
- [  ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

**Description of the changes being introduced by the pull request**:
Updates go version to 1.18.

Do we want to continue to test previous go versions?

**Please verify and check that the pull request fulfills the following requirements**:

- [ ] Tests have been added for the bug fix or new feature
- [ x ] Docs have been added for the bug fix or new feature

### CODE REVIEW

**Comment by asraa** [comment]
@rdimitrov does this address https://github.com/theupdateframework/go-tuf/issues/301 with the content in `TESTING.md`? Mostly my incompatibility is due to python-tuf being at version .20. We should update that too

**Comment by znewman01** [comment]
> Mostly my incompatibility is due to python-tuf being at version .20. We should update that too

#228 should do that (issue #221). It was blocked on #232, which #310 should fix. Once #310 is in I'll rebase

**Comment by asraa** [comment]
> https://github.com/theupdateframework/go-tuf/pull/228 should do that (issue https://github.com/theupdateframework/go-tuf/issues/221). It was blocked on https://github.com/theupdateframework/go-tuf/issues/232, which https://github.com/theupdateframework/go-tuf/pull/310 should fix. Once https://github.com/theupdateframework/go-tuf/pull/310 is in I'll rebase

Thank you! That's perf, closed that issue.

**Comment by rdimitrov** [comment]
> @rdimitrov does this address #301 with the content in `TESTING.md`? Mostly my incompatibility is due to python-tuf being at version .20. We should update that too

It does! Thank you! 🙏

**Review by rdimitrov — APPROVED** [review]

**Review by znewman01 — APPROVED** [review]

File: docs/TESTING.md [hunk 1]
```diff
@@ -1,3 +1,18 @@
 # Testing
 
-TODO([#301](https://github.com/theupdateframework/go-tuf/issues/301))
+The [Python interoperability tests](client/python_interop/) require Python 3
+(available as `python` on the `$PATH`) and the [`python-tuf`
+package](https://github.com/theupdateframework/python-tuf) installed. To use the correct vesions of the packages, it is recommended to use a virtual environment and install the dependencies via:
```
[the remark this answers is not in the document]
Comment (asraa) [reply docs/TESTING.md]: done!

**Review by znewman01 — APPROVED** [review]

**Review by rdimitrov — APPROVED** [review]

**Pull request merged**